src/StoredEvents/Models/EloquentStoredEventQueryBuilder.php

```diff
@@ -6,7 +6,11 @@
 use Spatie\EventSourcing\StoredEvents\StoredEvent;
 
 /**
- * @method \Spatie\EventSourcing\StoredEvents\Models\EloquentStoredEventCollection get
+ * @template TEloquentStoredEvent of EloquentStoredEvent
+ *
+ * @method \Spatie\EventSourcing\StoredEvents\Models\EloquentStoredEventCollection<EloquentStoredEvent> get()
```

Comment:
I tried to pass `TEloquentStoredEvent` as a type to the `EloquentStoredEventCollection`, but unfortunately, it is not possible. The best alternative solution is to pass `EloquentStoredEvent` as the type instead.